dashboard/serializers.py

```diff
@@ -0,0 +1,52 @@
+"""
+Provides functionality for serializing a ProgramEnrollment for the ES index
+"""
+from decimal import Decimal
+
+
+class UserProgramSerializer:
+    """
+    Provides functions for serializing a ProgramEnrollment for the ES index
+    """
+    @staticmethod
+    def calculate_certificate_grade_average(certificates):
+        """
+        Calculates an average grade (integer) from a list of certificates with <1 decimal grades
+        """
+        return None if len(certificates) == 0 else round(
+            (sum([Decimal(certificate['grade']) for certificate in certificates]) / len(certificates)) * 100
+        )
+
+    @classmethod
+    def serialize_valid_edx_data(cls, related_edx_obj_set, program):
+        """
+
+        Args:
+            related_edx_obj_set: Related QuerySet of cached edX model objects (eg: CachedEnrollment)
```

Comment:
1. `related_edx_obj_set` is kind of awkward... would something like this suffice?
`related_edx_obj_set (Many-to-one RelatedManager): QuerySet for cached edX model objects (eg: CachedEnrollment)

2. i think `program: Program object` should either stay how it is, or become `program (Program):` with nothing after the colon. anything more would be redundant and i don't see the use

3. this just returns a list of dictionaries, really. can you link me to that edx style guide again? i don't really know the best way to express that